Prox/Prox/PlaceCarousel/PlaceCarouselViewController.swift

```diff
@@ -16,6 +16,16 @@ protocol PlaceDataSource: class {
     func index(forPlace: Place) -> Int?
     func fetchPlace(placeKey: String, withEvent eventKey: String, callback: @escaping (Place?) -> ())
     func sortPlaces(byLocation location: CLLocation)
+
+    /// Returns all nearby places filtered by the given set of filters.
+    /// This function does not modify the data source.
+    func filterPlaces(filters: [PlaceFilter]) -> [Place]
+
+    /// Refreshes the data source by filtering with the enabled filters.
+    func refresh()
+
+    /// You must call refresh() after changing the enabled state of any filter!
```

Comment:
I guess I didn't think through it. :P I'd imagine it'd be ^ or id-lookup, but that indirection adds some complexity so maybe it's not worth it.